src/System.Drawing.Common/src/System/Drawing/Image.cs

```diff
@@ -46,7 +48,52 @@ public object Tag
             set => _userData = value;
         }
 
-        internal Image() { }
+        private protected Image() { }
+
+#pragma warning disable CA2229 // Implement Serialization constructor
+        private protected Image(SerializationInfo info, StreamingContext context)
+#pragma warning restore CA2229
+        {
+            byte[] dat = (byte[])info.GetValue("Data", typeof(byte[])); // Do not rename (binary serialization)
+
+            try
+            {
+                InitializeFromStream(new MemoryStream(dat));
+            }
+            catch (ExternalException e)
```

Comment:
Nit, you could collapse all this repeated code into one catch with an exception filter.

Incidentally I'm not sure whether it's worth keeping the Debug.Fail -- we try to avoid having asserts that can be triggered by bad inputs. In fact, you'll probably want to have at least one test that has garbage data, given all that's happening in InitializeFromStream, so you can't have asserts.

Reply:
we have several places in System.Drawing.Common where we use Debug.Fail. I think as this is just a port from netfx/Mono it's better to leave it as it is (at least for now). Though,  I tried the exception filtering approach and it still looked clunky.

I agree, we should have a negative serialization test here in System.Drawing.Common. If deserialization is not successful properties won't be set, so we can assert on them.

Reply:
> I think as this is just a port from netfx/Mono it's better to leave it as it is (at least for now)

Won't you be adding tests that would cause at least some of these catch blocks to be hit and end up triggering an assert?

Reply:
I tried to modify the blob and trigger these exceptions but failed. I'm open for suggestions how to hit these code paths. I don't understand why some of these catch blocks even exist.